api/src/main/java/io/jsonwebtoken/lang/Maps.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2019 jsonwebtoken.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.jsonwebtoken.lang;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility class to help with the manipulation of working with Maps.
+ * @since 0.11.0
+ */
+public final class Maps {
```

Comment:
Shouldn't this be a builder of sorts? Or in addition?

For example, I consider this a lot nicer than a bunch of parameters in a single method:

`Maps.put(key1, value1).put(key2, value2).put(key3, value3).build()`

(for example).

Thoughts?

Reply:
I'm not against it that 🤔 I just went the Java 11-ish route of `Map.of()`, 
I've done something things with a `.put()` in the past, my concern is Map returns the value type on a `put`, so that might be confusing?

I will say, I like the chained method look better than the overloaded method, mostly because it seems DRYer?

I'm not sure if `Map.of()` will ever see common use (at least for a while)

Reply:
ah, right - instead of `put`, then maybe `of` or `with` or `and`.

Reply:
I like this:

`Maps.of("foo", "whatever").and(key2, value2).and(key3, value3)...`

Maps.of creates a builder instance, puts the first key/value and returns the builder.  And just adds the key/value and returns the builder.  Looks nice and fluent.

Thoughts?

Reply:
I like that!